generators/javascript/javascript_generator.ts

```diff
@@ -253,7 +265,7 @@ export class JavascriptGenerator extends CodeGenerator {
       // Don't collect comments for nested statements.
       for (let i = 0; i < block.inputList.length; i++) {
         if (block.inputList[i].type === inputTypes.VALUE) {
```

Comment:
Yes, it would be nice if Blockly had been written taking full advantage of normal OO inheritance patterns.  Alas, prior to ES6 it was sufficiently hard to do inheritance correctly that even those who were otherwise competent JS programmers frequently got this wrong—and without `Object.setPrototypeOf` it was actually impossible to correctly subclass built-in classes that were exotic (e.g. `Array`) or had extra slots (e.g. `Date`), so it's no surprise that older code bases often use mixins and type tags rather than inheritance.